hail/src/main/scala/is/hail/expr/ir/streams/EmitStream.scala

```diff
@@ -1091,6 +1092,7 @@ object EmitStream {
                     cb.define(LpullRight)
                     if (rightProducer.requiresMemoryManagementPerElement)
                       cb += rightProducer.elementRegion.clearRegion()
+                    cb.assign(pulledRight, true)
```

Comment:
If we initialize `c` to 1, then the first thing we do is pull from the right. That should remove the need for `pulledRight` and simplify the logic.

Reply:
I don't think that's true -- we still need a bit that indicates we have to pull from both streams.

Reply:
Ah, I see it now. Thanks!